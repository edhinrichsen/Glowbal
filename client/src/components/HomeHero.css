.home-hero {
  background-color: var(--background-color-primary);
  box-shadow: var(--inner-shadow);
  background-image: url(../hero.svg);
  background-position: bottom center;
  background-repeat: no-repeat;
  background-size: contain;
  height: 300px;

}
[data-theme='dark'] .home-hero {
  background-image: url(../hero-dark.svg);
}

.toggle-home-hero {
  position: relative;
  top: 30px;
  right: 50px;
}
.home-hero h1 {
  margin-top: 100px;


}

.rocket {
  padding-right: 5px;
  background: #fff;
  -webkit-background-clip: text;
  background-clip: text;
  -webkit-text-fill-color: transparent;
  text-shadow: var(--drop-shadow);
}

.toggle-home-hero .anticon {
  display: flex;
  align-content: center;
  justify-content: center;
  padding-top: 7px;
}
