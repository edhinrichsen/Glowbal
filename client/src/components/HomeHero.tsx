import React, { useState } from 'react';
import { useAuth0 } from '@auth0/auth0-react';

import { Button, Tooltip } from 'antd';

import Emoji from './Emoji';

import './HomeHero.css';

function HomeHero() {
  const { isAuthenticated, loginWithRedirect } = useAuth0();
  const [isOpen, setIsOpen] = useState(true);

  return (
    <>
      {isOpen ? (
        <div className="home-hero">
          {/* <div className="toggle-home-hero" onClick={() => setIsOpen(!isOpen)}> */}
          <Tooltip title="Hide" placement="bottom">
            <div
              className="toggle-home-hero exit-button"
              onClick={() => setIsOpen(!isOpen)}
            ></div>
          </Tooltip>
          {/* </div> */}

          <div
            className="home-CTA"
            style={{
              display: 'flex',
              flexDirection: 'column',
              alignItems: 'center',
              justifyContent: 'center',
              minHeight: 250,
              padding: "30px"
            }}
          >
            <h1 style={{fontSize: "3em"}} className="gradient">
              <Emoji symbol="🛰️" /><br /><strong>Glowbal</strong>
            </h1>

            <p> Showcase your best work by creating a profile and...</p>

            {!isAuthenticated ? (
              <Button type="primary" onClick={() => loginWithRedirect()}>
                {' '}
                <span className="rocket">
                  <Emoji symbol="🚀" />
                </span>{' '}
                <strong className="rocket">Get Discovered</strong>
              </Button>
            ) : null}
          </div>
        </div>
      ) : null}
    </>
  );
}

export default HomeHero;
