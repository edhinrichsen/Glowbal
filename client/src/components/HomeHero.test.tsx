import React from 'react';
import { shallow, mount } from 'enzyme';
import HomeHero from './HomeHero';

describe('Testing HomeHero', () => {
  it('renders without crashing', () => {
    const component = shallow(<HomeHero />);
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('hides when closed', () => {
    const component = mount(<HomeHero />);

    component
      .find({ className: 'toggle-home-hero exit-button' })
      .simulate('click');

    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
