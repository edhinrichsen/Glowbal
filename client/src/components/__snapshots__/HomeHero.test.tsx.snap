// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomeHero hides when closed 1`] = `<HomeHero />`;

exports[`Testing HomeHero renders without crashing 1`] = `
<Fragment>
  <div
    className="home-hero"
  >
    <Tooltip
      arrowPointAtCenter={false}
      autoAdjustOverflow={true}
      mouseEnterDelay={0.1}
      mouseLeaveDelay={0.1}
      placement="bottom"
      title="Hide"
      transitionName="zoom-big-fast"
    >
      <div
        className="toggle-home-hero exit-button"
        onClick={[Function]}
      />
    </Tooltip>
    <div
      className="home-CTA"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "minHeight": 250,
        }
      }
    >
      <h1>
        <Emoji
          symbol="📝"
        />
         
        <strong>
          Glowbal
        </strong>
      </h1>
      <p>
         Showcase your best work by creating a profile and...
      </p>
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
        type="primary"
      >
         
        <span
          style={
            Object {
              "paddingRight": 5,
            }
          }
        >
          <Emoji
            symbol="🚀"
          />
        </span>
         
        <strong>
          Get Discovered
        </strong>
      </Button>
    </div>
  </div>
</Fragment>
`;
